File: server/test/reader-test/src/component-test/java/com/richo/reader/test/mocks/InMemoryEventStore.java
package com.richo.reader.test.mocks;

import com.richodemus.reader.events.Event;
import io.reactivex.Observable;
import io.reactivex.subjects.ReplaySubject;

import javax.inject.Singleton;

@Singleton
public class InMemoryEventStore
		implements com.richodemus.reader.label_service.EventStore, com.richo.reader.subscription_service.EventStore, com.richodemus.reader.user_service.EventStore
{
	private ReplaySubject<Event> replaySubject = ReplaySubject.create();

	@Override
	public void add(final Event event)
	{
		replaySubject.onNext(event);
	}

	@Override
	public Observable<Event> observe()
	{
		return replaySubject;
	}
}

File: server/services/label-service/src/test/kotlin/com/richodemus/reader/label_service/InMemoryEventStore.kt
package com.richodemus.reader.label_service

import com.richodemus.reader.events.Event
import io.reactivex.Observable
import io.reactivex.subjects.ReplaySubject

class InMemoryEventStore : EventStore {
    private val replaySubject = ReplaySubject.create<Event>()

    override fun add(event: Event) = replaySubject.onNext(event)

    override fun observe(): Observable<Event> = replaySubject
}
